mp/src/game/client/neo/ui/neo_hud_health_thermoptic_aux.cpp

```diff
@@ -54,6 +55,12 @@ CNEOHud_HTA::CNEOHud_HTA(const char* pElementName, vgui::Panel* parent)
 	}
 
 	m_hFont = scheme->GetFont("NHudOCRSmall");
+	m_hFontBuildInfo = scheme->GetFont("Default");
+
+	if (!g_pVGuiLocalize->ConvertANSIToUnicode(GIT_HASH, m_wszBuildInfo, sizeof(m_wszBuildInfo)))
+	{
+		m_wszBuildInfo[0] = L'\0';
+	}
```

Comment:
`m_wszBuildInfo` size is too short causing it to call the `...[0] = '\0'`.
![githash](https://github.com/NeotokyoRebuild/neo/assets/15316579/515c0d88-3ea6-41b4-aece-19a490c0d739)

Reply:
This one's on Windows
I guess just give it a big enough size and get the strlen instead

Reply:
Looks like 8 chars on Windows, or just the specific hash? hmm

Reply:
@Rainyan ok I know what happened. Git short hash is also 8 chars on Linux, but `ConvertANSIToUnicode` differs behavior somehow between Windows and Linux, where Linux just truncates without error?

Reply:
Could you try:
```diff
diff --git a/mp/src/game/client/neo/ui/neo_hud_health_thermoptic_aux.cpp b/mp/src/game/client/neo/ui/neo_hud_health_thermoptic_aux.cpp
index d26f528..a984cdf 100644
--- a/mp/src/game/client/neo/ui/neo_hud_health_thermoptic_aux.cpp
+++ b/mp/src/game/client/neo/ui/neo_hud_health_thermoptic_aux.cpp
@@ -105,11 +105,11 @@ void CNEOHud_HTA::DrawBuildInfo() const
        // DIY text shadow for contrast
        surface()->DrawSetTextColor(COLOR_BLACK);
        surface()->DrawSetTextPos(x + 2, y + 2);
-       surface()->DrawPrintText(m_wszBuildInfo, ARRAYSIZE(m_wszBuildInfo) - 1);
+       surface()->DrawPrintText(m_wszBuildInfo, Q_UnicodeLength(m_wszBuildInfo));
 
        surface()->DrawSetTextColor(COLOR_WHITE);
        surface()->DrawSetTextPos(x, y);
-       surface()->DrawPrintText(m_wszBuildInfo, ARRAYSIZE(m_wszBuildInfo) - 1);
+       surface()->DrawPrintText(m_wszBuildInfo, Q_UnicodeLength(m_wszBuildInfo));
```

I think I was trying to be clever with a compile-time expression here, but it's not working nicely for some reason.

Reply:
> @Rainyan ok I know what happened. Git short hash is also 8 chars on Linux, but `ConvertANSIToUnicode` differs behavior somehow between Windows and Linux, where Linux just truncates without error (for me on Linux it's just missing a '7' on the end in `9f36b307`)?

I'm fairly confident `git log -1 --format=%h` is only 7 characters long (sans the null terminator), so not seeing the 8th character would be normal here.

Reply:
@Rainyan I think version/distro defaults is causing it to differ. For me on both Windows and Arch Linux, it been defaults to 8 characters long. Turns out `git config core.abbrev XX` can be used to adjust any length of git hash as well. Maybe truncate it to 7 size via cmake?

Reply:
@Rainyan I've put in https://github.com/NeotokyoRebuild/neo/pull/349/commits/ec2bdf522fec9e351c79659d11b9791bd9683cf8 to just truncate it to 7 max. Generally don't think there's configs less than that so it should work out I think.

Reply:
Ah. Yeah, truncating in CMake to 7 max might be the cleanest way if the string formatting functions aren't behaving consistently across systems. Also, sorry for unleashing this mess unto your PR, didn't expect the complications 😓